### ISSUE TITLE
DagsterInvalidSubsetError when asset exists

### ISSUE THREAD

**Issue by EtienneT** [dagster-io/dagster#21152]
### Dagster version

1.7.0

### What's the issue?

I am getting the following error message for a new asset I created.  This only happens in 1.7.0, not in 1.6.14.  The backfill worked fine, but now this happens even when I try to manually trigger materialization for the asset from the UI. So obviously the asset is in the definition since I can access it in the UI and I can manually trigger the materialization.

`dagster._core.errors.DagsterInvalidSubsetError: AssetKey(s) ['price_daily'] were selected, but no AssetsDefinition objects supply these keys. Make sure all keys are spelled correctly, and all AssetsDefinitions are correctly added to the Definitions.`

### Deployment type

Local

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by EtienneT** [comment]
Tested in 1.6.14 again and it happens periodically.  Now we are at 1.7.1 in production and it still happens periodically.

**Comment by danielgafni** [comment]
Happens for me with multiple assets.

```
AssetKey(s) [...] were selected, but no AssetsDefinition objects supply these keys
```

**re-executing the same Run works without any errors**

**Comment by EtienneT** [comment]
> Happens for me with multiple assets.
> 
> ```
> AssetKey(s) [...] were selected, but no AssetsDefinition objects supply these keys
> ```
> 
> **re-executing the same Run works without any errors**

Not the case for me unfortunately.

**Comment by danielgafni** [comment]
Seems like a really big issue!

 Around 50% of my Runs randomly fail and work after re-running (sometimes). 

Tagging my friend @tacastillo because why not

**Comment by danielgafni** [comment]
This is extremely weird. 

I have this problem with all my assets. 

Again, the Runs sometimes fails, and sometimes complete without any issues. I can re-execute a failed run and it would work just fine. 

I've even compared `dagster-run-...` pod yaml contents for the failed and successful runs and found no difference. 

Maybe this has to do something with communication/network issues between different Dagster components?

**Comment by EtienneT** [comment]
This bug is getting urgent, it is making our dagster instance completely useless.

**Comment by EtienneT** [comment]
I can't really share the code inside the asset unfortunately, but I can share the code for the asset definition.  But it doesn't have anything special compared to other assets that work fine in my pipeline:

```
@asset(
    partitions_def=DailyPartitionsDefinition(
        start_date="2017-01-01", timezone="US/Eastern"
    ),
    compute_kind="pandas",
    ins={
        "price_recent_daily": AssetIn(
            partition_mapping=TimeWindowPartitionMapping(
                allow_nonexistent_upstream_partitions=True
            )
        ),
        "price_archive_daily": AssetIn(
            partition_mapping=TimeWindowPartitionMapping(
                allow_nonexistent_upstream_partitions=True
            )
        ),
    },
    auto_materialize_policy=AutoMaterializePolicy.eager(
        max_materializations_per_minute=25
    ),
    backfill_policy=BackfillPolicy.single_run(),
    io_manager_key="duckdb",
    metadata={"partition_expr": "Time"},
    group_name="Pricing",
)
@capture_op_exceptions
def rtlmp_5min_tohourly_daily(
    context: AssetExecutionContext,
    price_recent_daily: pd.DataFrame,
    price_archive_daily: pd.DataFrame,
) -> pd.DataFrame:
    ...
```

@capture_op_exceptions is to integrate Sentry.  Code from this thread: https://github.com/dagster-io/dagster/discussions/11102

Recently I tried changing how I add assets to the definitions.  I tried using load_assets_from_modules, also load_assets_from_package_module, but always get the same problem.  Sometimes the problem go away for some time and then comes back (and I did not push any new version of the code in production).  The problem happens in dev, in production...

This is how I load my assets, from the "main" package which has several sub package and modules.  And the asset migrations which I import separately.

```
all_assets = load_assets_from_package_module(main) + [migrations]

def = Definitions(
    assets=all_assets,
    ...
```

I hope this helps.

**Comment by gibsondan** [comment]
Thanks @EtienneT  - is it also possible to include the full stack of the error? Is it always coming from the same stack trace when you see it?

**Comment by EtienneT** [comment]
Yes, here you go:

```
dagster._core.errors.DagsterInvalidSubsetError: AssetKey(s) ['price_daily'] were selected, but no AssetsDefinition objects supply these keys. Make sure all keys are spelled correctly, and all AssetsDefinitions are correctly added to the `Definitions`.
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_grpc\impl.py", line 153, in core_execute_run
    yield from execute_run_iterator(
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\execution\api.py", line 869, in __iter__
    yield from self.execution_context_manager.prepare_context()
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_utils\__init__.py", line 510, in generate_setup_events
    obj = next(self.generator)
          ^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\execution\context_creation_job.py", line 299, in orchestration_context_event_generator
    context_creation_data = create_context_creation_data(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\execution\context_creation_job.py", line 112, in create_context_creation_data
    job_def = job.get_definition()
              ^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\reconstruct.py", line 265, in get_definition
    return check.not_none(self.get_repository_definition()).get_maybe_subset_job_def(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\repository_definition\repository_definition.py", line 330, in get_maybe_subset_job_def
    return defn.get_subset(
           ^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\job_definition.py", line 760, in get_subset
    return self._get_job_def_for_asset_selection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\job_definition.py", line 786, in _get_job_def_for_asset_selection
    job_asset_graph = get_asset_graph_for_job(self.asset_layer.asset_graph, selection)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_job.py", line 290, in get_asset_graph_for_job
    selected_keys = selection.resolve(parent_asset_graph)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_selection.py", line 420, in resolve
    return self.resolve_inner(asset_graph, allow_missing=allow_missing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_selection.py", line 664, in resolve_inner
    return reduce(
           ^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_selection.py", line 667, in <genexpr>
    selection.resolve_inner(asset_graph, allow_missing=allow_missing)
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_selection.py", line 708, in resolve_inner
    return self.left.resolve_inner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_selection.py", line 897, in resolve_inner
    raise DagsterInvalidSubsetError(
```

**Comment by gibsondan** [comment]
@EtienneT @danielgafni  one other thing either of you can do (if it's an option - entirely understand if it is not) is email us the debug output file from a successful and failed run from right around the same time (particularly if there is one that failed originally but passed on retry that will tell us a lot) - there's an option in the UI to do this here: 
<img width="435" alt="image" src="https://github.com/dagster-io/dagster/assets/8451211/5744b51f-564b-48b6-b79a-24c60394b225">

If that's an option you can email the output files to support at dagsterlabs dot come and we'll be able to get more troubleshooting information from the run as it appears in the Dagster UI with all its metadata.

**Comment by gibsondan** [comment]
The other angle we are pursuing here is whether there could possibly be something non-deterministic about the partitions definitions in your asset graph? Dagster figures out which assets are able be grouped together into runs by looking at which partitions definitions exist in the graph, and it somewhat relies on those staying the same every time you load the code.

So for example this unrealistic code that randomly assigns assets to different partitions definitions reproduces that error:

```
import random

from dagster import DailyPartitionsDefinition, asset

partitions_def_1 = DailyPartitionsDefinition(start_date="2017-01-01", timezone="US/Eastern")

partitions_def_2 = DailyPartitionsDefinition(start_date="2018-01-01", timezone="US/Eastern")


@asset(partitions_def=(partitions_def_1 if random.randint(1, 10) % 2 == 1 else partitions_def_2))
def asset1():
    pass


@asset(partitions_def=(partitions_def_1 if random.randint(1, 10) % 2 == 1 else partitions_def_2))
def asset2():
    pass
```

I doubt you're doing anything like that, but if there was something that was, say, selectively changing which partitions definitions get used, or changing the start date of a partitions definition in some non-deterministic way, or omitting partitions definitions from the graph entirely in certain situations, that might be a clue. Once we can reproduce this problem ourselves in some way i'm confident we'll be able to get to the bottom of it quickly.

**Comment by danielgafni** [comment]
@gibsondan done!

**Comment by EtienneT** [comment]
@gibsondan I can't really send debug info from the project I am working on...

I always use the same partition definition for the assets.  I have one dynamically partitioned asset (migrations), but this asset works and is completely independent from all other assets (it runs data migrations).

The partition definition for all assets are always inline in the asset definition.  So they can't change from run to run.

**Comment by danielgafni** [comment]
I generate the partitions from a yaml file. 

The file is checked into the repo and can be updated on-demand by executing one of the Python files, but the updating code is behind a "if __name__ == "__main__" statement. 

I don't think the yaml file can change unless Dagster is executing source Python files during code location load (seems unlikely). 

I'll be able to validate this hypothesis in an hour or so (once u get back home)

**Comment by gibsondan** [comment]
Thanks @danielgafni - we sent you a script that if you can run on your definitions object a few times and send us the results to see if they change, will answer a lot of questions here we hope.

**Comment by smackesey** [comment]
@EtienneT we believe we have a fix in #21497-- does the problem location happen to contain multiple partitions definitions having the same start date but different end dates?

**Comment by EtienneT** [comment]
My asset price_daily has two dependencies.  price_archive_daily has an end date (end_date="2021-01-01"), but not price_recent_daily.  Could this be it?

I got other assets with similar setup and they don't have this problem though.  As soon as this fix is included in a release I would be willing to try it.

Thanks,

**Comment by smackesey** [comment]
If there are two partitions definitions in the repo with the same start date but that differ in their end date (e.g. one has an end date defined, the other doesn't, as in your example), that would explain it.

**Comment by jamiedemaria** [comment]
yes, it should be out in this week's release

**Comment by EtienneT** [comment]
Tested with the newest release and the problem seems to be gone here!  Letting @danielgafni test on his end before closing bug.

Thanks guys!

**Comment by EtienneT** [comment]
Ok I talked too fast...

Got the error again, but on a different asset this time.  I am using the latest version 1.7.4 and 0.23.4.

I was doing a backfill of some failed partitions from an asset.  140+ of them got through successfully, but then 4 had this problem:

```
dagster._core.errors.DagsterInvalidSubsetError: AssetKey(s) ['price_merged'] were selected, but no AssetsDefinition objects supply these keys. Make sure all keys are spelled correctly, and all AssetsDefinitions are correctly added to the `Definitions`.
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_grpc\impl.py", line 153, in core_execute_run
    yield from execute_run_iterator(
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\execution\api.py", line 869, in __iter__
    yield from self.execution_context_manager.prepare_context()
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_utils\__init__.py", line 511, in generate_setup_events
    obj = next(self.generator)
          ^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\execution\context_creation_job.py", line 299, in orchestration_context_event_generator
    context_creation_data = create_context_creation_data(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\execution\context_creation_job.py", line 112, in create_context_creation_data
    job_def = job.get_definition()
              ^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\reconstruct.py", line 265, in get_definition
    return check.not_none(self.get_repository_definition()).get_maybe_subset_job_def(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\repository_definition\repository_definition.py", line 330, in get_maybe_subset_job_def
    return defn.get_subset(
           ^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\job_definition.py", line 760, in get_subset
    return self._get_job_def_for_asset_selection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\job_definition.py", line 786, in _get_job_def_for_asset_selection
    job_asset_graph = get_asset_graph_for_job(self.asset_layer.asset_graph, selection)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_job.py", line 290, in get_asset_graph_for_job
    selected_keys = selection.resolve(parent_asset_graph)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_selection.py", line 420, in resolve
    return self.resolve_inner(asset_graph, allow_missing=allow_missing)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\work\ML\dagster\.venv\Lib\site-packages\dagster\_core\definitions\asset_selection.py", line 897, in resolve_inner
    raise DagsterInvalidSubsetError(
```

I have several assets that use start_date, end_date, end_offset.  Sometimes end_offset of 1, sometimes of 2, most of the time no end_offset.  I have several different start_date in my DailyPartitionsDefinitions in my project.

The problem is now happening to several other assets when before it was only one asset.

This seems to be happening less often, but still a huge problem.

**Comment by EtienneT** [comment]
Yeah this is happening consistently on my most important asset now.  I can't update to this version because the problem is much worse for my pipeline.

**Comment by gibsondan** [comment]
@EtienneT is it possible to run this script locally on your Definitions object a few times to help us identify which of your partitions might be causing the nondeterministic behavior? We would be looking for places where the important asset that you mention is getting mapped to different jobs.

for job_def in defs.get_all_job_defs():
    if not job_def.name.startswith("__ASSET_JOB"):
        continue

    print("JOB NAME: " + str(job_def.name))

    print(f"ASSET KEYS: {job_def.asset_layer.asset_graph.materializable_asset_keys}")
    print(f"PARTITIONS DEFINITION: {job_def.partitions_def._asdict()} : {job_def.partitions_def}")

**Comment by EtienneT** [comment]
@gibsondan here is the result of the script:

```
JOB NAME: __ASSET_JOB_0
ASSET KEYS: {AssetKey(['price_today']), AssetKey(['nodes']), AssetKey(['migrations'])}
PARTITIONS DEFINITION: {'partition_fn': None, 'name': 'migrations'} : Dynamic partitions: "migrations"
JOB NAME: __ASSET_JOB_1
ASSET KEYS: {AssetKey(['price_5min_exante_daily_archive']), AssetKey(['price_base_daily_archive']), AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['price_final_daily_archive'])}
PARTITIONS DEFINITION: {'start': DateTime(2017, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': DateTime(2021, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'fmt': '%Y-%m-%d', 'end_offset': 0, 'cron_schedule': '0 0 * * *'} : Daily, starting 2017-01-01 US/Eastern.
JOB NAME: __ASSET_JOB_10
ASSET KEYS: {AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['strategy_profits'])}
PARTITIONS DEFINITION: {'start': DateTime(2023, 11, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 1, 'cron_schedule': '0 0 * * *'} : Daily, starting 2023-11-01 US/Eastern. End offsetted by 1 partition.
JOB NAME: __ASSET_JOB_11
ASSET KEYS: {AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['strategy_api']), AssetKey(['strategy'])}
PARTITIONS DEFINITION: {'start': DateTime(2023, 11, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 2, 'cron_schedule': '0 0 * * *'} : Daily, starting 2023-11-01 US/Eastern. End offsetted by 2 partitions.
JOB NAME: __ASSET_JOB_12
ASSET KEYS: {AssetKey(['nodes']), AssetKey(['price_prelim']), AssetKey(['price_today'])}
PARTITIONS DEFINITION: {'start': DateTime(2023, 12, 2, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 0, 'cron_schedule': '0 0 * * *'} : Daily, starting 2023-12-02 US/Eastern.
JOB NAME: __ASSET_JOB_2
ASSET KEYS: {AssetKey(['price_5min_exante_monthly_zip_archive']), AssetKey(['price_base_monthly_zip_archive']), AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['price_final_monthly_zip_archive'])}  
PARTITIONS DEFINITION: {'start': DateTime(2017, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': DateTime(2021, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'fmt': '%Y-%m-%d', 'end_offset': 0, 'cron_schedule': '0 0 1 * *'} : Monthly, starting 2017-01-01 US/Eastern.
JOB NAME: __ASSET_JOB_3
ASSET KEYS: {AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['price_5min_tohourly_daily'])}
PARTITIONS DEFINITION: {'start': DateTime(2017, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 0, 'cron_schedule': '0 0 * * *'} : Daily, starting 2017-01-01 US/Eastern.
JOB NAME: __ASSET_JOB_4
ASSET KEYS: {AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['fuelmix_historical_yearly'])}
PARTITIONS DEFINITION: {'start': DateTime(2017, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 0, 'cron_schedule': '0 0 1 1 *'} : 0 0 1 1 *, starting 2017-01-01 US/Eastern.
JOB NAME: __ASSET_JOB_5
ASSET KEYS: {AssetKey(['price']), AssetKey(['nodes']), AssetKey(['price_today'])}
PARTITIONS DEFINITION: {'start': DateTime(2017, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 1, 'cron_schedule': '0 0 * * *'} : Daily, starting 2017-01-01 US/Eastern. End offsetted by 1 partition.
JOB NAME: __ASSET_JOB_6
ASSET KEYS: {AssetKey(['price_base']), AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['dart'])}
PARTITIONS DEFINITION: {'start': DateTime(2017, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 2, 'cron_schedule': '0 0 * * *'} : Daily, starting 2017-01-01 US/Eastern. End offsetted by 2 partitions.
JOB NAME: __ASSET_JOB_7
ASSET KEYS: {AssetKey(['price_final_daily']), AssetKey(['nodes']), AssetKey(['price_today'])}
PARTITIONS DEFINITION: {'start': DateTime(2021, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': -3, 'cron_schedule': '0 0 * * *'} : Daily, starting 2021-01-01 US/Eastern. End offsetted by -3 partitions.
JOB NAME: __ASSET_JOB_8
ASSET KEYS: {AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['price_5min_recent_daily'])}
PARTITIONS DEFINITION: {'start': DateTime(2021, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 0, 'cron_schedule': '0 0 * * *'} : Daily, starting 2021-01-01 US/Eastern.
JOB NAME: __ASSET_JOB_9
ASSET KEYS: {AssetKey(['price_base_daily_recent']), AssetKey(['nodes']), AssetKey(['price_today']), AssetKey(['price_base_daily_exante'])}
PARTITIONS DEFINITION: {'start': DateTime(2021, 1, 1, 0, 0, 0, tzinfo=Timezone('US/Eastern')), 'timezone': 'US/Eastern', 'end': None, 'fmt': '%Y-%m-%d', 'end_offset': 2, 'cron_schedule': '0 0 * * *'} : Daily, starting 2021-01-01 US/Eastern. End offsetted by 2 partitions.
```

**Comment by gibsondan** [comment]
@EtienneT is it possible to run that a few more times and see if the asset in question is mapped to a different job, and post that same output again if it is? That would give us a ton of useful information to squash this.

**Comment by EtienneT** [comment]
The problem is not happening consistently.  It stopped happening now.  The asset that caused problems is now working perfectly...  This was also the behavior before, but now the problem seems to happen less frequently.

Could this be that I am using negative and positive end offsets in some daily definitions?

I added the script to my __init__.py file where I create the definition.  Any way of catching the error in there and printing the info only when the error happens?  This way I could let this run locally in dev and log only when the error happens.

**Comment by EtienneT** [comment]
This has been running flawlessly for the past week on my dev machine...  Pushed this version to staging last night to see how it goes.  So far so good.

**Comment by EtienneT** [comment]
This has been running for two weeks without any problems.  Closing issue for now.

**Comment by milicevica23** [comment]
Hi maybe a hint from my side, i had the similar error and found out that while building a dbt project on fly i have changed the pwd path and this coused the error that i see an asset in the UI but in the run the asset is not found.

**Comment by AndreaGiardini** [comment]
I am seeing this issue in 1.6.14 (we are testing 1.6.14, but prod is still on 1.5.x) and it looks like there was a report already about the same exact problem: https://github.com/dagster-io/dagster/issues/20504

How can I upgrade prod to 1.6.14 and avoid this issue? Going from 1.5 to 1.7 is not really an option, right?

**Comment by C-Loftus** [comment]
I am seeing this issue on dagster 1.8.7  Not clear what is causing it and I am a bit at a loss. Some assets fail, most are fine.
I have all assets in my `Definitions()` variable and I can see the assets as expected in the UI. However, when I materialize them via the UI, it fails.  

- I have a simple dagster configuration, and the only behavior that is non-trivial is the use of `StaticPartitionsDefinition` later in my pipeline. (Those don't fail though)
- I am loading in only 10 assets  (all by specifying the functions in a list) and they are all in the same file as the `main.py` entrypoint. 
    
Even a very trivial config like this fails when materializes. I have confirmed it is loaded correctly on the UI. 
```yml
@asset()
def my_config():
    """my config"""
    pass

```

**Comment by garethbrickman** [comment]
@C-Loftus Could you please share code snippets for us to try an exact reproduction?

**Comment by C-Loftus** [comment]
Apologies; I think this is related to code reloading issues. I deleted my associated db for dagster runtime data and it now seems to work. (Although code reloading still doesn't work reliably for me) Perhaps on my end somehow. Will look into this further and if I can replicate provide a better example

**Comment by JoeNaso** [comment]
I am experiencing this issue on `dagster==1.10.10` with a very simple unpartitioned asset. It appears in the UI but results in a `dagster._core.errors.DagsterInvalidSubsetError` error at run time

**Comment by h-enui** [comment]
Dagster Version is 1.10.1.
When I tried to copy the Dagster University code to create my own original job, I encountered "DagsterInvalidSubsetError".

In University's code
`adhoc_request = dg.AssetSelection.assets(["adhoc_request"]) `

and I expected to be able to define it as a list, but when I put in the second item I got an Exception.
`adhoc_request = dg.AssetSelection.assets(["adhoc_request","my_asset"]) `

However, the below pattern was succeeded.
`adhoc_request = dg.AssetSelection.assets("adhoc_request","my_asset") `
`adhoc_request = dg.AssetSelection.assets(adhoc_request, my_asset) ` <- function

Perhaps this is happening in some type of Union.
I hope the above will help someone.
